Assets/MixedRealityToolkit/Providers/UnityInput/MouseController.cs

```diff
@@ -65,32 +65,40 @@ public void Update()
                 return;
             }
 
-            if (InputSource.Pointers[0].BaseCursor != null)
-            {
-                controllerPose.Position = InputSource.Pointers[0].BaseCursor.Position;
-                controllerPose.Rotation = InputSource.Pointers[0].BaseCursor.Rotation;
-            }
-
-            mouseDelta.x = -UInput.GetAxis("Mouse Y");
-            mouseDelta.y = UInput.GetAxis("Mouse X");
-            MixedRealityToolkit.InputSystem?.RaiseSourcePositionChanged(InputSource, this, mouseDelta);
-            MixedRealityToolkit.InputSystem?.RaiseSourcePoseChanged(InputSource, this, controllerPose);
-            MixedRealityToolkit.InputSystem?.RaiseSourcePositionChanged(InputSource, this, UInput.mouseScrollDelta);
-
             for (int i = 0; i < Interactions.Length; i++)
             {
+                
+
                 if (Interactions[i].InputType == DeviceInputType.SpatialPointer)
                 {
+                    // add mouse delta as rotation
+                    var mouseDeltaRotation = Vector3.zero;
+                    mouseDeltaRotation.x += -UInput.GetAxis("Mouse Y");
+                    mouseDeltaRotation.y += UInput.GetAxis("Mouse X");
+                    
+                    var mouseInputProfile = MixedRealityToolkit.Instance?.GetService<MouseDeviceManager>()?.MouseInputProfile;
```

Comment:
GetService is surprisingly expensive, it does a whole bunch of allocations and linear string search. Typical usage is to have an internal ref and lazy-initialize it once, e.g.: [https://github.com/microsoft/MixedRealityToolkit-Unity/blob/acee452b82ffbfe6f48c0c4b1086058496d94161/Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/SolverHandler.cs#L145](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/acee452b82ffbfe6f48c0c4b1086058496d94161/Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/SolverHandler.cs#L145)

Reply:
caching profile now in controller and pointer